components/Image.js

```diff
@@ -23,7 +23,7 @@ class Image extends Component {
 
     // defaultSource is not supported on Android, so we manually
     // reassign the defaultSource prop to source if source is not defined
-    if ((Platform.OS === 'android') && (!source || !source.uri)) {
+    if (Platform.OS === 'android' && !source) {
```

Comment:
The original condition was here to handle cases when image URL may be undefined. For example, you often have an API response with optional URLs. If we removed the `!source.url` part of the condition then you will have to do something like this in those cases:
```
<Image source={user.avatarUrl ? { uri: user.avatarUrl } : undefined} />
```
instead of simply doing it like:
```
<Image source={{ uri: user.avatarUrl }} />
```

To support both use cases, we should detect if the source is a `require` call, and explicitly handle that case. It looks like the `require` will return a number, based on the prop types from react-native `Image`: https://github.com/facebook/react-native/blob/master/Libraries/Image/Image.android.js#L94-L95.